### PR TITLE
[Fleet] Fix docker publish step in buildkite

### PR DESCRIPTION
**Pull request by nchaulet**
## Description 

Fix docker publish step in buildkite

The if condition to build the docker image was not met 
Screenshot 2023-07-24 at 1 00 56 PM

It seems according to buildkite doc that `BUILDKITE_PULL_REQUEST ` should be false if it's not a pull request and not an empty string

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -105,6 +105,8 @@ steps:
       env:
         SERVICE_COMMIT_HASH: ${BUILDKITE_COMMIT:0:12}
         REMOTE_SERVICE_CONFIG: https://raw.githubusercontent.com/elastic/serverless-gitops/main/gen/gpctl/fleet/config.yaml
+    depends_on:
+      - step: "publish"
```
Comment (nchaulet) [inline .buildkite/pipeline.yml:109]: I added a dependency here so we do not update tags with non valid image

**Review by michel-laterman — APPROVED** [review]
lgtm

**Comment by nchaulet** [comment]
@michel-laterman should we replace the `fleet-server/pr-merge` as required by `buildkite/fleet-server ` ? How and who can do that?

**Comment by jlind23** [comment]
@nchaulet I just fixed this by adding buildkite as a required step and removed the jenkins one.

**Comment by jlind23** [comment]
Merging this with admin permissions as fleet-server jenkins build is blocking this but should not be required anymore.

**Pull request merged**